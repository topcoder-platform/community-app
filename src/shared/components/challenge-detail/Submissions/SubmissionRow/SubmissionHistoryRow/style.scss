@import "~styles/mixins";

.container {
  border-bottom: 1px solid #e9e9e9;

  @include xs-to-sm {
    padding-top: 10px;
    padding-bottom: 16px;
  }

  .row {
    font-size: 13px;
    color: $tc-gray-70;
    display: flex;
    border-bottom: 1px solid $tc-gray-10;
    padding: 16px 0;

    &.no-border {
      border: none;
    }

    @include xs-to-sm {
      padding: 0;
      display: flex;
      flex-direction: column;
    }
  }

  .col {
    text-align: center;

    @include roboto-medium;

    font-weight: 500;
    font-size: 14px;
    line-height: 22px;
    padding: 0;

    &.child {
      height: 57px;

      @include xs-to-sm {
        height: auto;
      }
    }

    &.col-1 {
      flex: 30;

      @include roboto-medium;

      color: #2a2a2a;
      font-weight: 500;
      font-size: 14px;
      line-height: 22px;
      text-align: left;

      @include xs-to-sm {
        flex: 10;
      }
    }

    &.col-2 {
      flex: 30;

      @include roboto-medium;

      color: #2a2a2a;
      font-weight: 500;
      font-size: 14px;
      line-height: 22px;
      text-align: left;

      @include xs-to-sm {
        display: block;
      }
    }

    &.col-3 {
      flex: 32;

      @include roboto-medium;

      color: #2a2a2a;
      font-weight: 500;
      font-size: 14px;
      line-height: 22px;
      text-align: left;
    }

    &.col-4 {
      flex: 10;

      @include roboto-medium;

      color: #2a2a2a;
      font-weight: 500;
      font-size: 14px;
      line-height: 22px;
      text-align: left;

      &.mm {
        flex: 28;
      }
    }

    &.center {
      text-align: center;

      @include xs-to-sm {
        text-align: left;
      }
    }
  }
}

.mobile-header {
  display: none;

  @include xs-to-sm {
    display: block;
    margin-top: 16px;
    font-size: 11px;
    line-height: 14px;

    @include barlow-bold;

    font-weight: 600;
    color: #767676;
    text-align: left;
  }
}

button {
  cursor: pointer;
  border: none;
  outline: none;
  font: inherit;
  color: inherit;
  padding: 0;
  background: transparent;
  text-transform: uppercase;
}
