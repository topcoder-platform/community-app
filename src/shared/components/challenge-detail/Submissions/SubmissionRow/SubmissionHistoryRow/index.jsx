/* eslint jsx-a11y/no-static-element-interactions:0 */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/**
 * SubmissionHistoryRow component.
 */

import React from 'react';
import PT from 'prop-types';
import moment from 'moment';
// import Completed from '../../../icons/completed.svg';
import Failed from '../../../icons/failed.svg';
import InReview from '../../../icons/in-review.svg';
import Queued from '../../../icons/queued.svg';

import './style.scss';

export default function SubmissionHistoryRow({
  isMM,
  submission,
  finalScore,
  provisionalScore,
  submissionTime,
  isReviewPhaseComplete,
  onShowPopup,
  submissionId,
  status,
  member,
}) {
  const getInitialReviewResult = () => {
    if (provisionalScore && provisionalScore < 0) return <Failed />;
    switch (status) {
      case 'completed':
        return provisionalScore;
      case 'in-review':
        return <InReview />;
      case 'queued':
        return <Queued />;
      case 'failed':
        return <Failed />;
      default:
        return provisionalScore;
    }
  };
  return (
    <div styleName="container">
      <div styleName="row no-border">
        { isMM ? <div styleName="col-1 col child" /> : null }
        <div styleName="col-2 col child">
          {submission}
        </div>
        <div styleName="col-3 col">
          <div styleName="col child">
            {(isMM || (!finalScore && finalScore !== 0)) || !isReviewPhaseComplete ? '-' : finalScore}
          </div>
          <div styleName="col child">
            {getInitialReviewResult()}
          </div>
        </div>
        <div styleName={`col-4 col history-time ${isMM ? 'mm' : ''}`}>
          <div styleName="col child">
            {moment(submissionTime).format('DD MMM YYYY')} {moment(submissionTime).format('HH:mm:ss')}
          </div>
        </div>
        {
          isMM && (
            <div styleName="col-5 col">
              <div
                role="button"
                tabIndex={0}
                styleName="col child"
                onClick={() => onShowPopup(true, submissionId, member)}
              >
                View Details
              </div>
            </div>
          )
        }
      </div>
    </div>
  );
}

SubmissionHistoryRow.defaultProps = {
  finalScore: null,
  provisionalScore: null,
  isReviewPhaseComplete: false,
};

SubmissionHistoryRow.propTypes = {
  member: PT.string.isRequired,
  isMM: PT.bool.isRequired,
  submission: PT.number.isRequired,
  finalScore: PT.number,
  status: PT.string.isRequired,
  provisionalScore: PT.number,
  submissionTime: PT.string.isRequired,
  isReviewPhaseComplete: PT.bool,
  submissionId: PT.string.isRequired,
  onShowPopup: PT.func.isRequired,
};
