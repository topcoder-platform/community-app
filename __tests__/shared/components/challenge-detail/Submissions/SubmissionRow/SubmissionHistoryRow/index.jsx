import React from 'react';
// import ReactDOM from 'react-dom';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import SubmissionHistoryRow from 'components/challenge-detail/Submissions/SubmissionRow/SubmissionHistoryRow';

const mockData = {
  isMM: true,
  submission: 1,
  finalScore: 80,
  provisionalScore: 80,
  submissionTime: '2017-11-06T15:49:35.000Z',
  isReviewPhaseComplete: false,
  status: 'completed',
  numWinners: 1,
  challengeStatus: 'Completed',
  auth: {
    tokenV3: 'tokenV3',
  },
  submissionId: '1',
  isLoggedIn: true,
};

describe('Matches shallow shapshot', () => {
  test('shapshot 1', () => {
    const renderer = new Renderer();

    renderer.render((
      <SubmissionHistoryRow {...mockData} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return <SubmissionHistoryRow {...this.props} />;
  }
}

describe('render properly', () => {
  test('click', () => {
    const instance = TU.renderIntoDocument((<Wrapper {...mockData} />));
    const matches = TU.scryRenderedDOMComponentsWithTag(instance, 'button');
    expect(matches).toHaveLength(1);
    TU.Simulate.click(matches[0]);
  });
});
