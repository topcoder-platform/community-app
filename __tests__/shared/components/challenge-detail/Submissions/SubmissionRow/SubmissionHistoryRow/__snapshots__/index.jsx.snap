// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot shapshot 1 1`] = `
<div
  className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__container___1WzuZ"
>
  <div
    className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__row___tH5eB src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__no-border___1tHzW"
  >
    <div
      className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__col-1___3g3Tt src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__col___2-hjE"
    >
      <div
        className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__mobile-header___3NsEQ"
      >
        SUBMISSION
      </div>
      <span>
        1
      </span>
    </div>
    <div
      className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__col-2___2KjRa src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__col___2-hjE"
    >
      <div
        className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__mobile-header___3NsEQ"
      >
        FINAL SCORE
      </div>
      <div>
        N/A
      </div>
    </div>
    <div
      className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__col-3___3MsR- src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__col___2-hjE"
    >
      <div
        className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__mobile-header___3NsEQ"
      >
        PROVISIONAL SCORE
      </div>
      <div>
        80
      </div>
    </div>
    <div
      className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__col-4___SJw2x src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__col___2-hjE src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__mm___2y_Fx"
    >
      <div
        className="src-shared-components-challenge-detail-Submissions-SubmissionRow-SubmissionHistoryRow-___style__mobile-header___3NsEQ"
      >
        TIME
      </div>
      <div>
        06 Nov 2017
         
        15:49:35
      </div>
    </div>
  </div>
</div>
`;
